test/unit/presenters/filtered_editions_presenter_test.rb

```diff
@@ -0,0 +1,112 @@
+# frozen_string_literal: true
+
+require "test_helper"
+
+class FilteredEditionsPresenterTest < ActiveSupport::TestCase
+  context "#editions" do
+    should "return all editions when no filters are specified" do
+      draft_guide = FactoryBot.create(:guide_edition, state: "draft")
+      published_guide = FactoryBot.create(:guide_edition, state: "published")
+
+      filtered_editions = FilteredEditionsPresenter.new.editions
+
+      assert_equal(2, filtered_editions.count)
+      assert_equal(draft_guide, filtered_editions[0])
+      assert_equal(published_guide, filtered_editions[1])
+    end
+
+    should "filter by specified states" do
+      draft_guide = FactoryBot.create(:guide_edition, state: "draft")
+      FactoryBot.create(:guide_edition, state: "published")
+
+      filtered_editions =
+        FilteredEditionsPresenter.new(states_filter: %w[draft]).editions
+
+      assert_equal(1, filtered_editions.count)
+      assert_equal(draft_guide, filtered_editions[0])
+    end
+
+    should "filter by 'assigned to'" do
+      anna = FactoryBot.create(:user, name: "Anna")
+      assigned_to_anna = FactoryBot.create(:guide_edition, assigned_to: anna.id)
+      FactoryBot.create(:guide_edition)
+
+      filtered_editions = FilteredEditionsPresenter.new(assigned_to_filter: anna.id).editions.to_a
+
+      assert_equal([assigned_to_anna], filtered_editions)
+    end
+
+    should "filter by 'not assigned'" do
+      anna = FactoryBot.create(:user, name: "Anna")
+      FactoryBot.create(:guide_edition, assigned_to: anna.id)
+      not_assigned = FactoryBot.create(:guide_edition)
+
+      filtered_editions =
+        FilteredEditionsPresenter.new(assigned_to_filter: "nobody").editions.to_a
+
+      assert_equal([not_assigned], filtered_editions)
+    end
+
+    should "ignore invalid 'assigned to'" do
+      anna = FactoryBot.create(:user, name: "Anna")
+      FactoryBot.create(:guide_edition, assigned_to: anna.id)
+      FactoryBot.create(:guide_edition)
+
+      filtered_editions =
+        FilteredEditionsPresenter.new(assigned_to_filter: "not a valid user id").editions
```

Comment:
could be single line similar to line 34

Reply:
This line would be too long, so I have left it as-is.